samples/matter/common/src/watchdog/watchdog.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include <cstdint>
+
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/slist.h>
+
+namespace Nrf
+{
+
+/**
+ * @brief Watchdog peripheral that can be use to detect code blocking and infinite loops
+ *
+ * It contains a possibility to create a different Watchdog sources that must be fed within the
+ *
+ * To start the global Watchdog use the Enable() method.
+ * This enables and starts all Watchdog source counters and from this point all Watchdog sources must be fed
+ * within their feeding windows.
+ *
+ * Watchdog sources cannot be disabled separately after enabling the Global Watchdog, but you can disable all of
+ * them at once using the Disable() static method. After that to re-enable Watchdog sources you need to call
+ * Restore() method on each of them and call Enable() method to start the Global Timer once again.
+ *
+ * Similarly to creating the objects, you must call Restore() method before calling the Enable() method.
+ */
+namespace Watchdog
+{
+	constexpr static size_t kMaximumAvailableSources = 8;
```

Comment:
nit: Is this something we should make configurable with Kconfig? Is this limitation coming from the watchdog peripheral? Maybe we can add a comment to explain it.

Reply:
This is a limitation that coming from the watchdog so I will add a note here.